bokeh/models/sources.py

```diff
@@ -138,23 +137,6 @@ def _data_from_df(df):
             new_data["index"] = index.tolist()
         return new_data
 
-    @classmethod
-    @deprecated("Bokeh 0.9.3", "ColumnDataSource initializer")
-    def from_df(cls, data):
-        """ Create a ``dict`` of columns from a Pandas DataFrame,
-        suitable for creating a ColumnDataSource.
-
-        Args:
-            data (DataFrame) : data to convert
-
-        Returns:
-            dict(str, list)
-
-        """
-        import warnings
-        warnings.warn("Method deprecated in Bokeh 0.9.3")
-        return cls._data_from_df(data)
-
     def to_df(self):
```

Comment:
It apparently gets used in charts/_attributes.py:
https://github.com/bokeh/bokeh/blob/master/bokeh/charts/_attributes.py#L112
and in crossfilter/models.py and plugins.py

I know odo also uses in to support bokeh as a backend too (ColumnDataSource -> df)

Whether any actual users use to_df, I doubt it?